src/main/java/com/amazon/opendistroforelasticsearch/security/resolver/IndexResolverReplacer.java

```diff
@@ -692,20 +692,7 @@ private boolean getOrReplaceAllIndices(final Object request, final IndicesProvid
             ((UpdateRequest) request).index(newIndices.length!=1?null:newIndices[0]);
         } else if (request instanceof SingleShardRequest) {
             final SingleShardRequest<?> gr = (SingleShardRequest<?>) request;
-            final String[] indices = gr.indices();
-            final String index = gr.index();
-
-            final List<String> indicesL = new ArrayList<String>();
-
-            if (index != null) {
-                indicesL.add(index);
-            }
-
-            if (indices != null && indices.length > 0) {
-                indicesL.addAll(Arrays.asList(indices));
-            }
-
```

Comment:
Out of curiosity, why the old logic adds both `index` and `indices` to `indecL`? But why the new logic ignores `index`?

Reply:
Check the link to [SingleShardRequest.java](https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/action/support/single/shard/SingleShardRequest.java#L97) shared in the PR description. It will be the same behavior. Don't know why old code was written that way hence removing it